File: APIGame/GameRuntime.cs
using System.Collections.Concurrent;

class GameRuntime
{
    public GameRuntime()
    {
        Castles = GameSetup.CreateCastles();
        Rocks = GameSetup.CreateRocks();
        StaticMap = GameSetup.CreateStaticMap(Castles, Rocks);
    }

    public Castle[] Castles { get; }
    public Rock[] Rocks { get; }
    public int[,] StaticMap { get; }
    public ConcurrentDictionary<Guid, ClientConnection> Connections { get; } = new();
    public ConcurrentDictionary<string, Tank> Tanks { get; } = new(StringComparer.OrdinalIgnoreCase);
    public List<Bullet> Bullets { get; } = new();
    public object GameLock { get; } = new();
    public CancellationTokenSource BroadcasterCts { get; } = new();

    long _bulletIdCounter = 0;
    public long NextBulletId() => Interlocked.Increment(ref _bulletIdCounter);
}

File: APIGameJava/src/main/java/com/example/apigame/game/GameRuntime.java
package com.example.apigame.game;

import com.example.apigame.model.Entities;
import org.springframework.stereotype.Component;

import java.util.List;
import java.util.Map;
import java.util.concurrent.ConcurrentHashMap;
import java.util.concurrent.atomic.AtomicLong;

@Component
public class GameRuntime {
    private final Entities.Castle[] castles;
    private final Entities.Rock[] rocks;
    private final int[][] staticMap;
    private final Map<String, Entities.Tank> tanks = new ConcurrentHashMap<>();
    private final Map<String, Entities.ClientConnection> connections = new ConcurrentHashMap<>();
    private final List<Entities.Bullet> bullets = new java.util.concurrent.CopyOnWriteArrayList<>();
    private final Object gameLock = new Object();
    private final AtomicLong bulletIdCounter = new AtomicLong();

    public GameRuntime() {
        this.castles = GameSetup.createCastles();
        this.rocks = GameSetup.createRocks();
        this.staticMap = GameSetup.createStaticMap(castles, rocks);
    }

    public Entities.Castle[] getCastles() { return castles; }
    public Entities.Rock[] getRocks() { return rocks; }
    public int[][] getStaticMap() { return staticMap; }
    public Map<String, Entities.Tank> getTanks() { return tanks; }
    public Map<String, Entities.ClientConnection> getConnections() { return connections; }
    public List<Entities.Bullet> getBullets() { return bullets; }
    public Object getGameLock() { return gameLock; }

    public long nextBulletId() {
        return bulletIdCounter.incrementAndGet();
    }
}
